include/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#----------------------------------------------------------------------
+# Public includes
+
+# Interface target to allow headers to be "linked" into other targets.
+add_library(headers INTERFACE)
```

Comment:
When a project has a library target of compiled source, there isn't usually a separate library for the headers.

If there was no compilation, I'd say yes, need this, but this isn't the case here.

Also... no headers are added as source to this target.

Reply:
Interesting.  I was trying to use the "best practice" of using `INTERFACE` targets where possible (so a source target can link to it and get the public header include path added).  

I guess it's dubious if the `OpenAssetIO::headers` target would ever be useful to consumers, but I had in my mind to boost, which exports a `Boost::headers` target, though in their case for good reason.

Happy to scrap this and move to the source target.

Reply:
Yeah boost is indeed different, with some header only libraries, so it has to have it.

I would keep things simple. Don't need a user trying to consume this target in isolation and finding nothing works.
Definitely move it all to the source target.

Reply:
Moved the public header config alongside the target config in 99d122c0